### PR TITLE
Add Edge to FeatureDetection

### PR DESCRIPTION
**Pull request by lilleyse**
Checks the `navigator.userAgent` to see if Edge is present. On Chrome, IE, and Firefox it is not.

I also changed the `isChrome` test because Edge contains Chrome in its `userAgent`.

I may have to fix `isSafari` as well for the same reason, but it might be ok since it also checks `/ Version\/([\.0-9]+)/` which would result in null on Edge. I'll test on Monday.

### CODE REVIEW

File: Source/Core/FeatureDetection.js [hunk 1]
```diff
@@ -29,11 +29,12 @@ define([
     function isChrome() {
         if (!defined(isChromeResult)) {
             isChromeResult = false;
-
-            var fields = (/ Chrome\/([\.0-9]+)/).exec(theNavigator.userAgent);
-            if (fields !== null) {
-                isChromeResult = true;
-                chromeVersionResult = extractVersion(fields[1]);
+            if (/Google Inc/.test(theNavigator.vendor)) {
```
Comment (mramato) [inline Source/Core/FeatureDetection.js]: I don't believe this check is correct, it will not work for all variants of Chrome (Chromium, Android, etc...).  Probably better to change it back and also just add a check for `Edge` to make sure we aren't on Edge masquerading as Chrome.

Comment (mramato) [inline Source/Core/FeatureDetection.js]: Basically add a `!isEdge()` here similar to what Safari does below.

**Comment by mramato** [comment]
Just update the `isSafari` check to have a `!isEdge()` in it, since Edge includes Safari in the UA.

**Comment by mramato** [comment]
Also, just a kindly reminder to everyone that we should avoid UA sniffing at all costs and only resort to it when there are no other options.  polygonOffset support is unfortunately one of these cases for the time being.

**Comment by lilleyse** [comment]
Updated.

**Comment by mramato** [comment]
Thanks, I'll merge as soon as this goes green.
